File: Тренировка 1.0/02. Линейный поиск/D. Больше своих соседей/Больше своих соседей.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    vector<int> list;
    do
    {
        int number;
        cin >> number;
        list.push_back(number);
    } while (cin.get() != '\n');

    if (list.size() <= 2) cout << 0;
    else
    {
        int count = 0;
        for (int i = 1, n = list.size(); i < n - 1; i++)
            if (list[i] > list[i - 1] && list[i] > list[i + 1]) count++;
        
        cout << count;
    }
}

File: Тренировка 1.0/02. Линейный поиск/D. Больше своих соседей/Больше своих соседей.cs
using System;

string[] neighbours = Console.ReadLine().Split(' ');
if (neighbours.Length == 1) Console.WriteLine(0);
else
{
    int count = 0;
    for (int i = 1; i < neighbours.Length - 1; i++)
    {
        int number = int.Parse(neighbours[i]);
        if (number > int.Parse(neighbours[i - 1]) && number > int.Parse(neighbours[i + 1])) count++;
    }
    Console.WriteLine(count);
}